src/packagedcode/pypi.py

```diff
@@ -414,76 +416,32 @@ def assign_package_to_resources(cls, package, resource, codebase, package_adder)
                 # relative paths need special treatment
                 # most of thense are references to bin ../../../bin/wheel
                 cannot_resolve = False
-                ref_resource = None
+                ref_resource = site_packages
                 while path_ref.startswith('..'):
                     _, _, path_ref = path_ref.partition('../')
-                    ref_resource = site_packages.parent(codebase)
+                    ref_resource = ref_resource.parent(codebase)
                     if not ref_resource:
                         cannot_resolve = True
                         break
+
                 if cannot_resolve or not ref_resource:
                     # TODO:w e should log these kind of things
                     continue
                 else:
-                    if package_uid:
+                    ref_resource = codebase.get_resource(
+                        path=os.path.join(ref_resource.path, path_ref)
+                    )
+                    if ref_resource and package_uid:
                         package_adder(package_uid, ref_resource, codebase)
             else:
-                ref_resource = get_resource_for_path(
-                    path=path_ref,
-                    root=site_packages,
-                    codebase=codebase,
+                # These are absolute paths from the site-packages directory
+                ref_resource = codebase.get_resource(
+                    path=os.path.join(site_packages.path, path_ref)
```

Comment:
We are always posix here.
```suggestion
                    path=posixpath.join(site_packages.path, path_ref)
```

Reply:
Added, thanks!